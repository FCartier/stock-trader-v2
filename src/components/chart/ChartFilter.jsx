import React from "react";
import { connect } from "react-redux";
import { chartFilter } from "./chartActions";

class ChartFilter extends React.Component {
  filter = event => {
    this.props.chartFilter(this.props.input, event.target.value);
  };

  render() {
    return (
      <div className="chart-filter">
        <button onClick={this.filter} value={"1d"}>1d</button>
        <button onClick={this.filter} value={"1m"}>1m</button>
        <button onClick={this.filter} value={"6m"}>6m</button>
        <button onClick={this.filter} value={"1y"}>1y</button>
        <button onClick={this.filter} value={"5y"}>5y</button>
      </div>
    );
  }
}

export default connect(
  state => ({
    input: state.input
  }),
  { chartFilter }
)(ChartFilter);
