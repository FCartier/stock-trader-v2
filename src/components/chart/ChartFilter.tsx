import React, { ReactElement } from "react";
import { connect } from "react-redux";
import { chartFilter } from "./chartActions";

interface IChartFilter {
   chartFilter: (input: string, event: any) => void,
   input: string
}

interface IState {
  input: string;
}

class ChartFilter extends React.Component<IChartFilter> {
  public filter = (event: any) => {
    this.props.chartFilter(this.props.input, event.target.value);
  };

  public render() {
    return (
      <div className="chart-filter">
        <button onClick={this.filter} value={"1d"}>1d</button>
        <button onClick={this.filter} value={"1m"}>1m</button>
        <button onClick={this.filter} value={"6m"}>6m</button>
        <button onClick={this.filter} value={"1y"}>1y</button>
        <button onClick={this.filter} value={"5y"}>5y</button>
      </div>
    );
  }
}

export default connect(
  (state : IState) => ({
    input: state.input
  }),
  { chartFilter }
)(ChartFilter);
