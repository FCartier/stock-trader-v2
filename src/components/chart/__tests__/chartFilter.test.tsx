import { configure, shallow, mount } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import * as React from "react";
import { ChartFilter } from "../ChartFilter";

configure({ adapter: new Adapter() });

describe("chart component", () => {
  test("chart filter component should render properly", () => {
    const component = shallow(
      <ChartFilter chartFilter={jest.fn()} input="AAPL" />
    );
    expect(component).toMatchSnapshot();
  });

  test("chart filter component should call chartFilter on click", () => {
    const chartFilter = jest.fn()
    const component = mount(
      <ChartFilter chartFilter={chartFilter} input="AAPL" />
    );

    component.find("button[value='6m']").simulate("click")
    expect(chartFilter).toHaveBeenCalledWith("AAPL", "6m")
  });
});