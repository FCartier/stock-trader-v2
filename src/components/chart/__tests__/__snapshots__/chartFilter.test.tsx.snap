// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chart component chart filter component should render properly 1`] = `
<Styled(div)>
  <button
    onClick={[Function]}
    value="1d"
  >
    1d
  </button>
  <button
    onClick={[Function]}
    value="1m"
  >
    1m
  </button>
  <button
    onClick={[Function]}
    value="6m"
  >
    6m
  </button>
  <button
    onClick={[Function]}
    value="1y"
  >
    1y
  </button>
  <button
    onClick={[Function]}
    value="5y"
  >
    5y
  </button>
</Styled(div)>
`;
